delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/ElasticSearchFiles.scala

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model
+
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.CatsEffectsClasspathResourceUtils
+import io.circe.syntax.EncoderOps
+import io.circe.{Json, JsonObject}
+
+trait ElasticSearchFiles {
+
+  /**
+    * Default elasticsearch mapping for a view
+    */
+  def defaultElasticsearchMapping: DefaultMapping
+
+  /**
+    * Default elasticsearch settings for a view
+    */
+  def defaultElasticsearchSettings: DefaultSettings
+
+  def emptyResults: EmptyResults
+
+  /** Mapping for the event metrics index */
+  def metricsMapping: MetricsMapping
+
+  /** Settings for the event metrics index */
+  def metricsSettings: MetricsSettings
+}
+
+final case class DefaultMapping(value: JsonObject)
+final case class DefaultSettings(value: JsonObject)
+final case class EmptyResults(value: Json)
+final case class MetricsMapping(value: JsonObject)
+final case class MetricsSettings(value: JsonObject)
```

Comment:
Could also use [monix newtypes](https://newtypes.monix.io/) and opaque types once we go to Scala 3

Reply:
I agree it's not always right, [names are not type safety](https://lexi-lambda.github.io/blog/2020/11/01/names-are-not-type-safety/). But here it's pushing the potential mistake to one place (`ElasticSearchFiles`) rather than at the 5 use sites - I actually broke a test due to this 😂 (it's probably a me problem, but when I see two parameters of the same type alarm bells go off)